engineering/conventions-go.md

```diff
@@ -0,0 +1,50 @@
+
+# Go Development Conventions
+
+## Introduction
+
+This guide documents development conventions for Go at source{d}. Check [general conventions](conventions.md) for language-independent conventions that are also applicable for Go projects.
+
+## Supported Go Versions
+
+* Support latest two stable major versions of Go (e.g. 1.9.x and 1.10.x).
+  * Some of our projects may support more versions, specially when there is wide-adoption outside source{d} (e.g. [go-git](https://github.com/src-d/go-git) currently supports three).
+
+## Dependency Management
+
+* If your project is an application:
+  * Use [dep](https://github.com/golang/dep) to manage dependencies.
+  * Check in the `vendor` directory into git.
+* If your project is a library:
+  * You may use [dep](https://github.com/golang/dep) to manage dependencyes if needed. If you do, do not commit `vendor` directory.
+
+## Build System
+
+* If your project requires fetching or building dependencies, implement that in the `dependencies` rule.
+* If binaries can be built, do so in the `packages` rule. Note that [src-d/ci](https://github.com/src-d/ci) provides that rule by default.
+
+## Testing
+
+* Use [testify](https://github.com/stretchr/testify).
+* Use [suites](https://github.com/stretchr/testify#suite-package) to share setup/teardown code among related tests. When testing multiple methods of the same struct, you generally want to start with a suite.
+
+## Executables
+
+* Put sources for your executable commands in `cmd/<your command>`.
+* Use [go-flags](https://github.com/jessevdk/go-flags) for CLI options parsing.
+
+## Error handling
+
+* Use [src-d/go-errors](https://github.com/src-d/go-errors).
+
+## Logging
+
+* Use logging in your applications.
+* Use [logrus](https://github.com/sirupsen/logrus) for logging.
+* Use [fields](https://github.com/sirupsen/logrus#fields) instead of string formatting in the log message.
+
+## Best Practices
+
+* Use `gofmt`.
+* Use `go vet`.
```

Comment:
What about [golint](https://github.com/golang/lint)? It is a good reflection of Go Code Review Comments document mentioned below.

Reply:
I'd recommend https://github.com/dominikh/go-tools/tree/master/cmd/megacheck

Reply:
Ok. I checked with some people and indeed `golint` is used extensively during development, even indirectly, because vscode uses it by default. I'm adding both since we're actually using them.

Reply:
I think vim-go go uses it.